### PR TITLE
feat: add OAuth2 with Auth0, add JWT auth with backend, update solidstart

### PR DESCRIPTION
**Pull request by williambl**
hells its finally done :feelsgood: 

This removes old login/signup stuff, defers to Auth0 for that stuff instead. It also adds a new way of communicating with the backend - the solidstart server obtains a sigle JWT, which it uses to authenticate with the backend. The server then makes requests on behalf of the browser clients.

There are now multiple env variables that must be set. Their names are documented in the README. Their values have been sent to you on discord.

This requires mondaylunch/gatos-backend#82 to be merged to work properly.

### CODE REVIEW

**Comment by insertish** [comment]
:trollface:

File: packages/website/vite.config.ts [hunk 1]
```diff
@@ -3,4 +3,7 @@ import { defineConfig } from "vite";
 
 export default defineConfig({
   plugins: [solid()],
+  server: {
+    port: 5173
```
Comment (LectureMonk) [inline packages/website/vite.config.ts:7]: Explicit Port bad

**Review by LectureMonk — CHANGES_REQUESTED** [review]

File: packages/website/vite.config.ts (hunk 1 above)
Comment (williambl) [reply packages/website/vite.config.ts:7]: Without this it hosts on port 3000 by default on my machine.

**Review by LectureMonk — APPROVED** [review]

**Comment by williambl** [comment]
uh

**Pull request merged**